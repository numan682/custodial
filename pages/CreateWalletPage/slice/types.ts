/* --- STATE --- */
export interface CreateWalletState {
  mnemonic: string[]
  checkbox: boolean
}
