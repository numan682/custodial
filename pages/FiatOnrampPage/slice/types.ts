export interface FiatOnrampState {
  thirdPartyAcknowledged: boolean
}
