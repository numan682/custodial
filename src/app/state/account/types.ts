import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
