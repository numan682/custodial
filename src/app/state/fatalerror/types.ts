/* --- STATE --- */
export interface FatalErrorPayload {
  message: string
  stack?: string
  sagaStack?: string
}

export interface FatalErrorState {
  error?: FatalErrorPayload
}
