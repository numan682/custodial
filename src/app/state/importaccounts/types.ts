import { ErrorPayload } from 'types/errors'
import { WalletType } from '../wallet/types'
import { BalanceDetails } from '../account/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance?: BalanceDetails
  path: number[]
  pathDisplay: string
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  Idle = 'idle',
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  accountsSelectionPageNumber: number
  step: ImportAccountsStep
}
