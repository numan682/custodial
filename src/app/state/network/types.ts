/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string

  /** Current epoch */
  epoch: number

  /** Minimum staking amount */
  minimumStakingAmount: number
}
