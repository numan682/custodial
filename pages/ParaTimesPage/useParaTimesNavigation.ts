import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { backend } from 'vendors/backend'
import { BackendAPIs } from 'config'
import { WalletType } from 'app/state/wallet/types'
import { selectType } from 'app/state/wallet/selectors'

export type ParaTimesNavigationHook = {
  canAccessParaTimesRoute: boolean
  getParaTimesRoutePath: (address: string) => string
  navigateToAmount: () => void
  navigateToDeposit: () => void
  navigateToParaTimes: () => void
  navigateToRecipient: () => void
  navigateToConfirmation: () => void
  navigateToSummary: () => void
  navigateToWithdraw: () => void
  paraTimesRouteLabel: string
}

export const useParaTimesNavigation = (): ParaTimesNavigationHook => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const walletType = useSelector(selectType)
  const canAccessParaTimesRoute = backend() === BackendAPIs.OasisScan && walletType !== WalletType.Ledger
  const getParaTimesRoutePath = (address: string) => `/account/${address}/paratimes`
  const navigateToDeposit = useCallback(() => dispatch(paraTimesActions.navigateToDeposit()), [dispatch])
  const navigateToWithdraw = useCallback(() => dispatch(paraTimesActions.navigateToWithdraw()), [dispatch])
  const navigateToParaTimes = useCallback(() => dispatch(paraTimesActions.navigateToParaTimes()), [dispatch])
  const navigateToRecipient = useCallback(() => dispatch(paraTimesActions.navigateToRecipient()), [dispatch])
  const navigateToAmount = useCallback(() => dispatch(paraTimesActions.navigateToAmount()), [dispatch])
  const navigateToConfirmation = useCallback(
    () => dispatch(paraTimesActions.navigateToConfirmation()),
    [dispatch],
  )
  const navigateToSummary = useCallback(() => dispatch(paraTimesActions.navigateToSummary()), [dispatch])
  const paraTimesRouteLabel = t('menu.paraTimes', 'ParaTimes')

  return {
    canAccessParaTimesRoute,
    getParaTimesRoutePath,
    navigateToAmount,
    navigateToConfirmation,
    navigateToDeposit,
    navigateToParaTimes,
    navigateToRecipient,
    navigateToSummary,
    navigateToWithdraw,
    paraTimesRouteLabel,
  }
}
