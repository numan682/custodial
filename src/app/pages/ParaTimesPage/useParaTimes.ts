import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { TFunction } from 'i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { selectSelectedNetwork, selectTicker } from 'app/state/network/selectors'
import { selectAccountAvailableBalance, selectAccountIsLoading } from 'app/state/account/selectors'
import { selectAddress } from 'app/state/wallet/selectors'
import { selectParaTimes } from 'app/state/paratimes/selectors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload, ExhaustedTypeError } from 'types/errors'
import { paraTimesConfig, RuntimeTypes, ParaTime, type ParaTimeConfig } from '../../../config'

type AvailableParaTimesForNetwork = {
  isEvm: boolean
  value: ParaTime
}

const evmcParaTimes = Object.keys(paraTimesConfig).filter(
  key => paraTimesConfig[key as ParaTime].type === RuntimeTypes.Evm,
)

export type ParaTimesHook = {
  accountAddress: string | undefined
  accountIsLoading: boolean
  availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[]
  balance: StringifiedBigInt | null
  balanceInBaseUnit: boolean
  clearTransactionForm: () => void
  isDepositing: boolean
  isEvmcParaTime: boolean
  isLoading: boolean
  isWalletEmpty: boolean
  paraTimeConfig: ParaTimeConfig
  paraTimeName: string
  setTransactionForm: (formValues: TransactionForm) => void
  submitTransaction: () => void
  ticker: string
  transactionError: ErrorPayload | undefined
  transactionForm: TransactionForm
  usesOasisAddress: boolean
}

const getParaTimeName = (t: TFunction, paraTime: ParaTime) => {
  switch (paraTime) {
    case ParaTime.Cipher:
      return t('paraTimes.common.cipher', 'Cipher')
    case ParaTime.Emerald:
      return t('paraTimes.common.emerald', 'Emerald')
    case ParaTime.Sapphire:
      return t('paraTimes.common.sapphire', 'Sapphire')
    default:
      throw new ExhaustedTypeError(
        t('paraTimes.validation.unsupportedParaTime', 'Unsupported ParaTime'),
        paraTime,
      )
  }
}

export const useParaTimes = (): ParaTimesHook => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const clearTransactionForm = useCallback(() => {
    dispatch(paraTimesActions.clearTransactionForm())
  }, [dispatch])
  const submitTransaction = useCallback(() => {
    dispatch(paraTimesActions.submitTransaction())
  }, [dispatch])
  const setTransactionForm = (formValues: TransactionForm) =>
    dispatch(paraTimesActions.setTransactionForm(formValues))
  const { balance, isLoading, transactionError, transactionForm } = useSelector(selectParaTimes)
  const accountBalance = useSelector(selectAccountAvailableBalance)
  const accountIsLoading = useSelector(selectAccountIsLoading)
  const accountAddress = useSelector(selectAddress)
  const selectedNetwork = useSelector(selectSelectedNetwork)
  const ticker = useSelector(selectTicker)
  const isDepositing = transactionForm.type !== TransactionTypes.Withdraw
  const isEvmcParaTime = evmcParaTimes.includes(transactionForm.paraTime!)
  const needsEthAddress = isDepositing && isEvmcParaTime
  const balanceInBaseUnit = isDepositing || (!isDepositing && !isEvmcParaTime)
  const paraTimeName = transactionForm.paraTime ? getParaTimeName(t, transactionForm.paraTime) : ''
  const availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[] = (
    Object.keys(paraTimesConfig) as ParaTime[]
  )
    .filter(paraTimeKey => paraTimesConfig[paraTimeKey][selectedNetwork].runtimeId)
    .sort((a, b) => paraTimesConfig[a].displayOrder - paraTimesConfig[b].displayOrder)
    .map(paraTimeKey => ({
      isEvm: paraTimesConfig[paraTimeKey].type === RuntimeTypes.Evm,
      value: paraTimeKey,
    }))
  const walletBalance = !isDepositing ? balance : accountBalance
  const paraTimeConfig = paraTimesConfig[transactionForm.paraTime!]

  return {
    accountAddress,
    accountIsLoading,
    availableParaTimesForSelectedNetwork,
    balance: walletBalance,
    balanceInBaseUnit,
    clearTransactionForm,
    isDepositing,
    isEvmcParaTime,
    isLoading,
    isWalletEmpty: walletBalance === '0',
    paraTimeConfig,
    paraTimeName,
    setTransactionForm,
    submitTransaction,
    ticker,
    transactionError,
    transactionForm,
    usesOasisAddress: !needsEthAddress,
  }
}
