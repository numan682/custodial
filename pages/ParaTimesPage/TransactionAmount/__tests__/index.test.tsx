import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { paraTimesActions } from 'app/state/paratimes'
import { TransactionForm } from 'app/state/paratimes/types'
import { useParaTimes, ParaTimesHook } from '../../useParaTimes'
import { useParaTimesNavigation, ParaTimesNavigationHook } from '../../useParaTimesNavigation'
import { TransactionAmount } from '..'
import { ParaTime } from 'config'

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}))
jest.unmock('react-i18next')
jest.mock('../../useParaTimes')
jest.mock('../../useParaTimesNavigation')

describe('<TransactionAmount />', () => {
  const mockUseParaTimesResult = {
    balance: '1000000000',
    balanceInBaseUnit: true,
    isDepositing: true,
    isEvmcParaTime: false,
    isWalletEmpty: false,
    paraTimeName: 'Cipher',
    paraTimeConfig: { gasPrice: 5n, feeGas: 500_000n, decimals: 9 },
    ticker: 'ROSE',
    transactionForm: {
      amount: '',
      recipient: 'dummyAddress',
      paraTime: ParaTime.Cipher,
      ethPrivateKey: 'ethPrivateKey',
    },
    usesOasisAddress: true,
  } as ParaTimesHook
  const mockUseParaTimesNavigationResult = {} as ParaTimesNavigationHook

  beforeEach(() => {
    jest.mocked(useParaTimes).mockReturnValue(mockUseParaTimesResult)
    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
  })

  it('should render component', () => {
    const { container } = render(<TransactionAmount />)

    expect(container).toMatchSnapshot()
  })

  it('should render EVMc withdraw variant component', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
      isEvmcParaTime: true,
      paraTimeName: 'Emerald',
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(screen.getByTestId('paraTime-content-description')).toMatchSnapshot()
  })

  it('should render collapsible Advance section', async () => {
    render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'Advanced' }))
    expect(screen.getByPlaceholderText('Fee Amount (nano ROSE)')).toBeVisible()
    expect(screen.getByPlaceholderText('Fee Gas')).toBeVisible()
    await userEvent.click(screen.getByRole('button', { name: 'Advanced' }))
    expect(screen.getByPlaceholderText('Fee Amount (nano ROSE)')).not.toBeVisible()
    expect(screen.getByPlaceholderText('Fee Gas')).not.toBeVisible()
  })

  it('should render disabled variant', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      balance: '0',
      isWalletEmpty: true,
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(screen.getByText('The account is empty. There is nothing to withdraw.')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('0')).toBeDisabled()
    expect(screen.getByRole('button', { name: 'MAX' })).toBeDisabled()
    expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled()
  })

  it('should set input value to max', async () => {
    const setTransactionForm = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({ ...mockUseParaTimesResult, setTransactionForm })
    const { rerender } = render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'MAX' }))
    expect(setTransactionForm).toHaveBeenNthCalledWith(1, expect.objectContaining({ amount: '1' }))

    jest
      .mocked(useParaTimes)
      .mockReturnValue({ ...mockUseParaTimesResult, setTransactionForm, balance: '1563114365108133939632' })
    rerender(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'MAX' }))
    expect(setTransactionForm).toHaveBeenNthCalledWith(
      2,
      expect.objectContaining({ amount: '1563114365108.133939632' }),
    )
  })

  it('should require amount field on form submit', async () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(screen.getByText('Field is required')).toBeInTheDocument()
    expect(navigateToConfirmation).not.toHaveBeenCalled()
  })

  it('should validate amount field on form submit', async () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '10000000000000',
      },
    } as ParaTimesHook)
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(screen.getByText('Insufficient balance')).toBeInTheDocument()
    expect(navigateToConfirmation).not.toHaveBeenCalled()
  })

  it('should validate amount and fee on form submit', async () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '1',
        feeAmount: '2500000',
      },
    } as ParaTimesHook)
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(screen.getByText('Insufficient balance to pay the fee')).toBeInTheDocument()
    expect(navigateToConfirmation).not.toHaveBeenCalled()
  })

  it('should navigate to confirmation step when amount is valid', async () => {
    const setTransactionForm = jest.fn()
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '1',
      } as TransactionForm,
      setTransactionForm,
    } as ParaTimesHook)
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    await userEvent.click(screen.getByRole('button', { name: 'Next' }))
    expect(setTransactionForm).toHaveBeenCalledWith(
      expect.objectContaining({ amount: '1', defaultFeeAmount: '0' }),
    )
    expect(navigateToConfirmation).toHaveBeenCalled()
  })

  it('should fetch non EVMc paraTime balance', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(mockDispatch).toHaveBeenCalledWith({
      payload: undefined,
      type: paraTimesActions.fetchBalanceUsingOasisAddress.type,
    })
  })

  it('should fetch EVMc paraTime balance', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
      isEvmcParaTime: true,
      transactionForm: {
        ...mockUseParaTimesResult.transactionForm,
        paraTime: ParaTime.Emerald,
      },
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(mockDispatch).toHaveBeenCalledWith({
      payload: undefined,
      type: paraTimesActions.fetchBalanceUsingEthPrivateKey.type,
    })
  })
})
