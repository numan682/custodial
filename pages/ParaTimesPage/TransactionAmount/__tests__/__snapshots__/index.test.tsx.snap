// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionAmount /> should render EVMc withdraw variant component 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<span
  class="c0"
  data-testid="paraTime-content-description"
>
  Please enter the amount of ROSE to withdraw and then click "Next"
</span>
`;

exports[`<TransactionAmount /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 96px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 384px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 24px;
  padding-bottom: 24px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

.c14 {
  font-size: 14px;
  line-height: 20px;
  font-weight: bolder;
}

.c15 {
  font-size: 12px;
  line-height: 18px;
  color: lightText;
  font-weight: 600;
}

.c11 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus > circle,
.c11:focus > ellipse,
.c11:focus > line,
.c11:focus > path,
.c11:focus > polygon,
.c11:focus > polyline,
.c11:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus::-moz-focus-inner {
  border: 0;
}

.c11:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible) > circle,.c11:focus:not(:focus-visible) > ellipse,
.c11:focus:not(:focus-visible) > line,.c11:focus:not(:focus-visible) > path,
.c11:focus:not(:focus-visible) > polygon,.c11:focus:not(:focus-visible) > polyline,
.c11:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c18 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  margin-bottom: 24px;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 20px;
  font-size: 14px;
  line-height: 20px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c18:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus > circle,
.c18:focus > ellipse,
.c18:focus > line,
.c18:focus > path,
.c18:focus > polygon,
.c18:focus > polyline,
.c18:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus::-moz-focus-inner {
  border: 0;
}

.c18:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible) > circle,.c18:focus:not(:focus-visible) > ellipse,
.c18:focus:not(:focus-visible) > line,.c18:focus:not(:focus-visible) > path,
.c18:focus:not(:focus-visible) > polygon,.c18:focus:not(:focus-visible) > polyline,
.c18:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c22 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  width: 100%;
}

.c22:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c22:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c22:focus > circle,
.c22:focus > ellipse,
.c22:focus > line,
.c22:focus > path,
.c22:focus > polygon,
.c22:focus > polyline,
.c22:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c22:focus::-moz-focus-inner {
  border: 0;
}

.c22:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c22:focus:not(:focus-visible) > circle,.c22:focus:not(:focus-visible) > ellipse,
.c22:focus:not(:focus-visible) > line,.c22:focus:not(:focus-visible) > path,
.c22:focus:not(:focus-visible) > polygon,.c22:focus:not(:focus-visible) > polyline,
.c22:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c22:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c10 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c10::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c10::-moz-placeholder {
  color: #AAAAAA;
}

.c10:-ms-input-placeholder {
  color: #AAAAAA;
}

.c10::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c10::-moz-focus-inner {
  border: none;
  outline: none;
}

.c10:-moz-placeholder,
.c10::-moz-placeholder {
  opacity: 1;
}

.c9 {
  position: relative;
  width: 100%;
}

.c1 {
  margin-bottom: 48px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c20 {
  -webkit-transition: max-height 200ms,opacity 200ms;
  transition: max-height 200ms,opacity 200ms;
  opacity: 0;
  overflow: hidden;
}

.c12 {
  position: absolute;
  top: 10px;
  right: 0;
  z-index: 2;
  background-color: #FFFFFF;
  font-weight: bold;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c21 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Deposit to ParaTime
    </h3>
    <div
      class="c2"
      style="max-width: 550px;"
    >
      <span
        class="c3"
        data-testid="paraTime-content-description"
      >
        Please enter the amount of ROSE to deposit and then click "Next"
      </span>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <form
          novalidate=""
        >
          <div
            class="c6"
          >
            <div
              class="c7"
              style="position: relative;"
            >
              <div
                class="c6 "
                style="width: 100%;"
              >
                <div
                  class="c8 "
                >
                  <div
                    class="c9"
                  >
                    <input
                      autocomplete="off"
                      class="c10"
                      inputmode="decimal"
                      name="amount"
                      placeholder="0"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <button
                class="c11 c12"
                type="button"
              >
                MAX
              </button>
            </div>
            <div
              class="c13"
            >
              <span
                class="c14"
              >
                <span>
                  Available:
                   
                </span>
                <span>
                  <span>
                    1.0
                  </span>
                  <span
                    class="c15"
                  >
                    <span
                      class="notranslate"
                      translate="no"
                    >
                       undefined
                    </span>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <div
            class="c16"
          >
            <div
              class="c17"
            >
              <button
                class="c18"
                type="button"
              >
                Advanced
              </button>
            </div>
            <div
              aria-hidden="true"
              class="c19 c20"
            />
          </div>
          <div
            class="c21"
          >
            <div
              class="c5"
            >
              <button
                class="c22"
                type="submit"
              >
                Next
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
